### PR TITLE
Add random_state class

### PR DESCRIPTION
**Pull request by LukichevaPolina**
- Added RandomState class
- Added RandomState.uniform() method
- New tests implemented to cover both RandomState() and uniform() calls

### CODE REVIEW

**Review by densmirn — COMMENTED** [review]
Looks like we need to support the following use case:
```python
rs = dpnp.random.RandomState(seed=1, sycl_queue=dpctl.SyclQueue())
rs.uniform(lower=0., upper=1, size=10, usm_type='device')
```
Perhaps we also need to provide parameter `device` for `RandomState`.
@Alexander-Makaryev what do you think about extra parameter?

File: dpnp/random/dpnp_iface_random.py [hunk 1]
```diff
@@ -1566,7 +1566,7 @@ def uniform(low=0.0, high=1.0, size=None):
         else:
             if low > high:
                 low, high = high, low
-            return dpnp_rng_uniform(low, high, size, dtype=numpy.float64).get_pyobj()
+            return dpnp_rng_uniform(low, high, size, dtype=numpy.float64, usm_type=usm_type).get_pyobj()
```
[the remark this answers is not in the document]
Comment (LukichevaPolina) [reply dpnp/random/dpnp_iface_random.py:1569]: In this case the following error will happen:
`SyntaxError: positional argument follows keyword argument`

[the remark this answers is not in the document]
Comment (densmirn) [reply dpnp/random/dpnp_iface_random.py:1569]: Function `dpnp_rng_uniform` doesn't have keyword arguments at all. So we need to call the function using only positional arguments. Please pay you attention to parameter `dtype`.

File: dpnp/backend/kernels/dpnp_krnl_random.cpp [hunk 2]
```diff
@@ -2133,6 +2135,8 @@ DPCTLSyclEventRef dpnp_rng_uniform_c(DPCTLSyclQueueRef q_ref,
 
     sycl::queue q = *(reinterpret_cast<sycl::queue*>(q_ref));
 
+    mt19937_class* mt19937 = reinterpret_cast<mt19937_class*>(mt19937_in);
```
[the remark this answers is not in the document]
Comment (LukichevaPolina) [reply dpnp/backend/kernels/dpnp_krnl_random.cpp:2138]: When casting to IRandomState type, field `engine` will have type void *.  Function mkl_rng::generate can`t take it us input. Should I reimplement interface class?

File: dpnp/random/dpnp_iface_random.py [hunk 3]
```diff
@@ -1711,3 +1716,7 @@ def zipf(a, size=None):
             return dpnp_rng_zipf(a, size).get_pyobj()
 
     return call_origin(numpy.random.zipf, a, size)
+
+
+seed = 1
+default_random_state = RandomState(seed)
```
Comment (densmirn) [inline dpnp/random/dpnp_iface_random.py:1722]: I would rename it to `dpnp_random_state` if patched the PR.

**Review by densmirn — APPROVED** [review]

**Comment by Alexander-Makaryev** [comment]
Building in CI is failed.

**Review by oleksandr-pavlyk — APPROVED** [review]
I think this is big improvement over previous state. 

Existing minor issues will be addressed in subsequent RPs.

Thanks @antonwolfy for finishing this work.

**Pull request merged**